ta/arch/arm/ta.ld.S

```diff
@@ -28,7 +28,11 @@ SECTIONS {
 	}
         .plt : { *(.plt) }
 
-	.eh_frame : { *(.eh_frame) }
+	.eh_frame_hdr : {
+		*(.eh_frame_hdr)
+		*(.eh_frame_entry .eh_frame_entry.*)
+	}
+	.eh_frame : { KEEP(*(.eh_frame)) *(.eh_frame.*) }
```

Comment:
Is the `KEEP()` directive mandatory?

Reply:
I think so. It is present in the GCC default linker script (see output of `arm-linux-gnueabihf-gcc -Wl,-verbose`) and it means the symbols in the specified input section have to be kept even when `--gc-sections` is given and the symbols are not referenced.

Now I must admit there is a bit of cargo-cult programming here, because I can't justify why `KEEP()` is not equally applied to `.eh_frame.*` ;-)

Reply:
> Now I must admit there is a bit of cargo-cult programming here, because I can't justify why KEEP() is not equally applied to .eh_frame.* ;-)

Aha, so that's what's smelling. ;-)

How about dropping this until something breaks and we know why it's needed?

Reply:
This comment I have not addressed. I think we'd rather trust GCC here, our usage is no different as far as I can tell. Applications that produce `.eh_frame` info during compilation need to have them linked into the final binary so that the C++ runtime can find them (using `dl_iterate_phdr()`).

Reply:
Note, I can confirm that if we don't add a `.eh_frame` output section to our linker script, C++ exceptions don't work. I was just commenting on the KEEP() thing.

Reply:
OK, thanks for looking into this.